crates/source_analyzer/src/structs.rs

```diff
@@ -0,0 +1,788 @@
+                            Some(a) => match a {
+                                StructHierarchy::Optional { child } => match &**child {
+                                    StructHierarchy::Struct { fields }
+                                        if fields.len() == 1
+                                            && match fields.get(&"b".to_string()) {
+                                                Some(b) => match b {
+                                                    StructHierarchy::Optional { child } => match &**child {
+                                                        StructHierarchy::Struct { fields }
+                                                            if fields.len() == 1
+                                                                && match fields.get(&"c".to_string()) {
+                                                                    Some(c) => match c {
+                                                                        StructHierarchy::Optional { child } => match &**child {
+                                                                            StructHierarchy::Field {
+                                                                                data_type: matched_data_type,
+                                                                            } if &data_type
+                                                                                == matched_data_type =>
+                                                                            {
+                                                                                true
+                                                                            }
+                                                                            _ => false,
+                                                                        },
+                                                                        _ => false,
+                                                                    },
+                                                                    None => false,
+                                                                } =>
+                                                            true,
+                                                        _ => false,
+                                                    },
+                                                    _ => false,
+                                                },
+                                                None => false,
+                                            } =>
+                                        true,
+                                    _ => false,
+                                },
+                                _ => false,
+                            },
+                            None => false,
+                        } =>
+                    true,
+                _ => false,
+            },
+            "{hierarchy:?} does not match"
+        );
```

Comment:
These pyramids are very hard to read and verify. Ideally one would define a StructHierarchy and then match/eq it against the generated one.

Sadly btreemaps aren't very ergonomic to declare with content since there is no `vec![]` equivalent as far as I can see.
But perhaps it's still better to declare each element and then `BTreeMap::insert` them together?

Reply:
Hopefully improved.